src/v/cluster/log_eviction_stm.h

```diff
@@ -32,18 +35,31 @@ class consensus;
  * adjust that offset with _stm_manager->max_collectible_offset(), write the
  * raft snapshot and notify the storage layer that log eviction can safely
  * proceed up to the adjusted offset.
+ *
```

Comment:
This commit would benefit from more detailed commit message, it changes multiple interfaces and abstractions would be great to know what is the reasoning